Engine/DataFeeds/Enumerators/DataQueueOptionChainUniverseDataCollectionEnumerator.cs

```diff
@@ -0,0 +1,155 @@
+        }
+
+        /// <summary>
+        /// Returns current option chain enumerator position
+        /// </summary>
+        public OptionChainUniverseDataCollection Current { get; private set; }
+
+        /// <summary>
+        /// Returns current option chain enumerator position
+        /// </summary>
+        object IEnumerator.Current => Current;
+
+        /// <summary>
+        /// Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.
+        /// </summary>
+        public void Dispose()
+        {
+            Underlying.Dispose();
+        }
+
+        /// <summary>
+        /// Advances the enumerator to the next element of the collection.
+        /// </summary>
+        /// <returns>
+        /// true if the enumerator was successfully advanced to the next element; false if the enumerator has passed the end of the collection.
+        /// </returns>
+        public bool MoveNext()
+        {
+            Underlying.MoveNext();
+
+            if (Underlying.Current == null)
+            {
+                Current = null;
+                return true;
+            }
+
+            if (!_needNewCurrent)
+            {
+                // refresh on date change (in exchange time zone)
+                _needNewCurrent = _timeProvider.GetUtcNow().ConvertFromUtc(_subscriptionRequest.Configuration.ExchangeTimeZone).Date != _lastEmitTime.Date;
+            }
+
+            if (_needNewCurrent)
+            {
+                var localTime = _timeProvider.GetUtcNow()
+                    .RoundDown(_subscriptionRequest.Configuration.Increment)
+                    .ConvertFromUtc(_subscriptionRequest.Configuration.ExchangeTimeZone);
+
+                // loading the list of futures contracts and converting them into zip entries
+                var symbols = _universeProvider.LookupSymbols(_subscriptionRequest.Security.Symbol.ID.Symbol, _subscriptionRequest.Security.Type);
+                var zipEntries = symbols.Select(x => new ZipEntryName { Time = localTime, Symbol = x } as BaseData).ToList();
+                _currentData = new OptionChainUniverseDataCollection
+                {
+                    Symbol = _subscriptionRequest.Security.Symbol,
+                    Underlying = Underlying.Current,
+                    Data = zipEntries,
+                    Time = localTime,
+                    EndTime = localTime
+                };
+
+                _lastEmitTime = localTime;
+
+                Current = _currentData;
+                _needNewCurrent = false;
+            }
+            else
+            {
+                if (Current == null)
+                {
+                    Current = _currentData;
+                }
+
+                Current.Underlying = Underlying.Current;
+                Current.Time = Underlying.Current.EndTime;
+                Current.EndTime = Underlying.Current.EndTime;
+
+                _lastEmitTime = Current.EndTime;
```

Comment:
The underlying enumerator is required with options because universe selection dynamic filters (such as +/- N strikes) need the updated price of the underlying instrument.
Basically when the underlying instrument is trading, the enumerator will return the same collection of contracts for the same date but the `Underlying` property will be updated every minute. When the date changes, this collection will be reloaded completely from the `IUniverseProvider`.

In the Futures case, we removed the underlying enumerator since we had previously removed the underlying (due to unsupported security types such as indices) and the enumerator wasn't actually used.